core/server/worker/src/main/java/alluxio/worker/netty/BlockReadHandler.java

```diff
@@ -149,71 +153,41 @@ private void openBlock(BlockReadRequestContext context, Channel channel) throws
       }
 
       do {
-        long lockId;
-        if (request.isPersisted()) {
-          lockId = mWorker.lockBlockNoException(request.getSessionId(), request.getId());
-        } else {
-          lockId = mWorker.lockBlock(request.getSessionId(), request.getId());
-        }
-        if (lockId != BlockLockManager.INVALID_LOCK_ID) {
-          try {
-            BlockReader reader = mWorker.createBlockReader(request.getSessionId(), request.getId(),
-                request.getStart(), false, request.getOpenUfsBlockOptions());
-            String metricName = "BytesReadAlluxio";
-            context.setBlockReader(reader);
-            context.setCounter(MetricsSystem.counter(metricName));
-            mWorker.accessBlock(request.getSessionId(), request.getId());
+        try {
+          BlockReader reader = mWorker.createBlockReader(request.getSessionId(), request.getId(),
+              request.getStart(), false, request.getOpenUfsBlockOptions());
+          String metricName = "BytesReadAlluxio";
+          context.setBlockReader(reader);
+          context.setCounter(MetricsSystem.counter(metricName));
+          mWorker.accessBlock(request.getSessionId(), request.getId());
+          if (reader.getChannel() instanceof FileChannel) {
```

Comment:
If not FileChannel, are we silently doing nothing is right?

Reply:
Exactly. If it is not FileChannel, we don't need to reset the position.